src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private static final double COEFFICIENT_OF_EXPANSION = 1.5;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[INITIAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        expand();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        expand();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() >= elements.length) {
+            expand();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```

I dont think we need this